app/models/minion.rb

```diff
@@ -54,6 +63,18 @@ def self.assign_roles!(roles: {}, default_role: :minion)
     end
   end
 
+  def self.select_masters(roles)
```

Comment:
Would it ever be desirable to select 2 minions explicitly, and have the rest end up as masters? I'm thinking `Minion.unassigned_role.pluck(:id)` can just be removed entirely, allowing the "assign default role if there is any minion left with no role" code to handle this instead.